### PR TITLE
SQL: Introduce Operation enum, replace bool

### PR DESCRIPTION
**Pull request by whyman**

### CODE REVIEW

File: src/database/sql_database.cc [hunk 1]
```diff
@@ -364,9 +364,14 @@ std::vector<std::shared_ptr<SQLDatabase::AddUpdateTable>> SQLDatabase::_addUpdat
...
     if (obj->getID() != INVALID_OBJECT_ID)
-        throw_std_runtime_error("tried to add an object with an object ID set");
+        throw_std_runtime_error("Tried to add an object with an object ID set");
 
-    std::vector<std::shared_ptr<SQLDatabase::AddUpdateTable>> tables = _addUpdateObject(obj, false, changedContainer);
+    if (config->getOption(CFG_SERVER_STORAGE_DRIVER) == "sqlite3") {
+        exec("BEGIN TRANSACTION");
+    } else {
+        exec("START TRANSACTION");
+    }
```
Comment (KarlStraussberger) [inline src/database/sql_database.cc]: You'd better add a new property beginTransaction and set it in constructor call - a virtual method seems a bit overdone.

File: src/database/sql_database.cc [hunk 2]
```diff
@@ -403,13 +409,13 @@ void SQLDatabase::updateObject(std::shared_ptr<CdsObject> obj, int* changedConta
         exec("START TRANSACTION");
     }
     for (const auto& addUpdateTable : data) {
-        std::string operation = addUpdateTable->getOperation();
+        Operation op = addUpdateTable->getOperation();
         std::unique_ptr<std::ostringstream> qb;
-        if (operation == "update") {
+        if (op == Operation::Update) {
```
Comment (KarlStraussberger) [inline src/database/sql_database.cc:411]: Switch ?

**Comment by KarlStraussberger** [comment]
Forget my first comment. :rocket:

File: src/database/sql_database.cc (hunk 2 above)
Comment (whyman) [reply src/database/sql_database.cc:411]: Good idea!

Comment (whyman) [reply src/database/sql_database.cc]: Done!

**Comment by whyman** [comment]
Should be ready for merge now

**Pull request merged**